### PR TITLE
HLM-3161:: Split Age into years and Months across the application

### PR DESCRIPTION
**Pull request by Ramkrishna-egov**
[HLM-3161](https://digit-discuss.atlassian.net/browse/HLM-3161)

[HLM-3161]: https://digit-discuss.atlassian.net/browse/HLM-3161?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      DateTime inputDate = inputFormat.parse(date);
+
+      return inputDate;
+    } on Exception catch (e) {
+      if (kDebugMode) {
+        print(e);
+      }
+
+      return null;
+      //return e on console
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart:34]: remove commented code

File: apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      }
+      //return error
+    }
+  }
+
+  static DateTime? getFormattedDateToDateTime(String date) {
+    try {
+      DateFormat inputFormat;
+      inputFormat = date.contains('-')
+          ? DateFormat('dd-MM-yyyy')
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart:23]: date formate should be passed as params

File: apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+      return null;
+      //return e on console
+    }
+  }
+
+  static String getDateFromTimestamp(int timestamp) {
+    DateTime date = DateTime.fromMillisecondsSinceEpoch(timestamp);
+
+    return DateFormat("dd/MM/yyyy").format(date);
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart:41]: +1

**Review by naveen-egov — CHANGES_REQUESTED** [review]

File: apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart (hunk 2 above)
Comment (Ramkrishna-egov) [reply apps/health_campaign_field_worker_app/lib/utils/date_conversions.dart:22]: @naveen-egov This accepts input as 'dd-MM-yyyy' or 'dd/MM/yyyy', Its checking which format the input is accordingly returning the dateTime object

**Review by naveen-egov — APPROVED** [review]

**Pull request merged**